### PR TITLE
Upgrade c-ares to 1.17.2

### PR DESCRIPTION
**Pull request by apolcyn**
As title

The latest c-ares release is actually 1.18.1, but 1.18.1 adds new dependencies on Windows APIs (https://docs.microsoft.com/en-us/windows/win32/api/netioapi/nf-netioapi-getunicastipaddresstable and https://docs.microsoft.com/en-us/windows/win32/api/netioapi/nf-netioapi-freemibtable in particular) which are causing link errors. Without solving those issues yet, upgrading to 1.17.2 is still useful.

Note - potentially breaking change:

Because this new version of c-ares changes directory of public headers as well as internal sources, this might break users who are building gRPC with cmake and older versions of a pre-installed c-ares library.

### CODE REVIEW

**Review by jtattermusch — APPROVED** [review]
LGTM as long as this doesn't affect our ability to install gRPC C++ with a pre-installed c-ares (since the path to c-ares headers seems to have changed).
I think our distribtests should test that scenario, but they haven't finished yet. Please doublecheck.

**Comment by apolcyn** [comment]
Bazel C/C++ dbg macos failure is a tool failure

File: .gitmodules [hunk 1]
```diff
@@ -23,7 +23,7 @@
 [submodule "third_party/cares/cares"]
 	path = third_party/cares/cares
 	url = https://github.com/c-ares/c-ares.git
-	branch = cares-1_12_0
+	branch = cares-1_17_1
```
Comment (akca) [inline .gitmodules:26]: @apolcyn should this be `cares-1_17_2` instead?

Comment (apolcyn) [reply .gitmodules:26]: you're right, thanks. Just fixed that

Comment (akca) [reply .gitmodules:26]: Say hi to my first (tiny-little) contribution to gRPC then 🕺 [#28780](https://github.com/grpc/grpc/pull/28780)

**Pull request merged**